jobrunner/executors/local.py

```diff
@@ -175,7 +175,9 @@ def execute(self, job_definition):
                 label=LABEL,
                 labels=get_job_labels(job_definition),
                 extra_args=extra_args,
+                volume_type=volume_api.volume_type,
```

Comment:
Rather than threading an extra arg through I think it would be worth considering making `volume` a more structured object i.e. dataclass or namedtuple with `source`, `target` and `type`.

Reply:
That is cleaner, but also potentially a non-trivial amount of work, as I think doing it properly like will likely mean refactoring the VolumeAPI and resulting tests?

I have two days left to try get this ready for production before our next initiative starts, or else I don't know when we'll get change to work on it again. And I am still working on the backend-server plumbing to consume this image.

So, is worth doing here and now? Or something for later?

Reply:
Oh yeah, totally ignorable for now! Sorry if I didn't make that clear. Just think it's always worth highlighting stuff when you happen to notice it.

Reply:
:heavy_check_mark: https://github.com/opensafely-core/job-runner/issues/748